## Project description
E COLI OUTBREAK 22.11.2020

The goal is to reproduce part of [E. coli O104:H4 Genome Analysis](https://github.com/ehec-outbreak-crowdsourced/BGI-data-analysis/wiki) on TY2482 sequencing data and answer following questions:
1. What is genomic sequence of E.Coli X?
2. What strain of E. coli is E. coli X most similar to? (Where did it come from?)
3. What are the genes that E. coli X contains?
4. Which of these genes make E. coli X distinct?
5. How did E. coli X evolve to obtain these genes?
6. How did E. coli X become pathogenic?

## Exploring the dataset
Runned `fasQC` on datasets.

| Data | Number of reads |
| ---  | ----------------|
| SRR292678 F | 5499346  |
| SRR292678 R | 5499346  |
| SRR292770 F | 5102041  |
| SRR292770 R | 5102041  |
| SRR292862 F | 5102041  |
| SRR292862 R | 5102041  |

## K-mer profiling
Runned `jellyfish` on `SRR292678` reads:
```
jellyfish count -m 31 -o SRR292678_F_31 -s 6M -t 8 -C SRR292678sub_S1_L001_R1_001.fastq
jellyfish histo -o SRR292678_F_31_hysto -t 8 SRR292678_F_31
```
Used R script to plot histos and estimate genome size:
```
install.packages("ggplot2")
library("ggplot2")
dataF <- read.csv("SRR292678_F_31_hysto", sep=" ")
dataR <- read.csv("SRR292678_R_31_hysto", sep=" ")

ggplot(dataF, aes(x=a, y=b)) + geom_bar(stat="identity") + scale_y_log10()
ggplot(dataR, aes(x=a, y=b)) + geom_bar(stat="identity") + scale_y_log10()

# looking for peak, peak is 62 for forward
ggplot(dataF, aes(x=a, y=b)) + geom_bar(stat="identity") + coord_cartesian(xlim=c(50, 75), ylim = c(75000, 100000))

# looking for peak, peak is 59 for reverse
ggplot(dataR, aes(x=a, y=b)) + geom_bar(stat="identity") + coord_cartesian(xlim=c(50, 75), ylim = c(85000, 90000))

# total k-mers
sum(as.numeric(dataF$a * dataF$b))/60.5

```

Estimated genome size is `5453897`.

## Assembling E.Coli X genome from paired reads
Used SPAdes. 

## Effect of read correction
Same script for counting genome size on corrected reads results `5155220`.
