## Project description
ANTIBIOTIC RESISTANCE 1.11.2020

The goals are to analyze that sequencing data from a strain of E. coli resistant to the antibiotic ampicillin to locate the mutations responsible for giving E. coli its antibiotic resistance property, to research the genes that are mutated to identify the mechanism of antibiotic resistance in each case, and to make recommendations for alternative antibiotics a doctor could use to treat each strain.

## Filtering the reads
Filter reads with following parameters:
* paired end reads, where `amp_res_1.fastq` is forward read, `amp_res_1.fastq` is reversed
* 4 threads as processor has 4 physical cores
* phred33, (because fastqc shows `Sanger / Illumina 1.9` encoding)
* with logging, 'cause always log everything
* cut bases from the start, if quality < 20
* cut bases from the end, if quality < 20
* use sliding window with window size 10, quality 20
* drop reads with length < 20
```
TrimmomaticPE -threads 4 -phred33 -trimlog ./logs/trimmomatic01112020.log amp_res_1.fastq amp_res_2.fastq -baseout amp_trimmed.fq SLIDINGWINDOW:10:20 LEADING:20 TRAILING:20 MINLEN:20
```

Results:
```
Input Read Pairs: 455876 Both Surviving: 445689 (97.77%) Forward Only Surviving: 9758 (2.14%) Reverse Only Surviving: 284 (0.06%) Dropped: 145 (0.03%)
```

Checked if number of reads matched manually with `wc -l`.

Runned `fastqc` on `amp_trimmed_1P.fc` and `amp_trimmed_2P.fc`, problem with low per sequence base quality was fixed.

## Aligning sequences to reference
### Indexing the reference file
```
bwa index GCF_000005845.2_ASM584v2_genomic.fna
```

### Aligning reads

`bwa mem ./reference_seq/GCF_000005845.2_ASM584v2_genomic.fna amp_trimmed/amp_trimmed_1P.fq amp_trimmed/amp_trimmed_2P.fq > alignment.sam`

### Compressing SAM file
Compressing to `.bam` file:
```
samtools view -S -b alignment.sam > alignment.bam
```
Getting basic stats:
```
samtools flagstat alignment.bam
```
Results:
```
891635 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 secondary
257 + 0 supplementary
0 + 0 duplicates
890569 + 0 mapped (99.88% : N/A)
891378 + 0 paired in sequencing
445689 + 0 read1
445689 + 0 read2
887530 + 0 properly paired (99.57% : N/A)
889384 + 0 with itself and mate mapped
928 + 0 singletons (0.10% : N/A)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)
```
